axelrod/cooperation.py

```diff
@@ -1,7 +1,66 @@
 from . import eigen
+from axelrod import Actions
 
-def cooperation_matrix(results):
+C, D = Actions.C, Actions.D
+
+# As yet unused until RoundRobin returns interactions
+def cooperation_matrix(interactions, nplayers):
+    """
+    The cooperation matrix from a single round robin.
+
+    Parameters
+    ----------
+    interactions : dictionary
+        A dictionary of the form:
+
+        e.g. for a round robin between Cooperator, Defector and Alternator
+        with 2 turns per round:
+        {
+            (0, 0): [(C, C), (C, C)].
+            (0, 1): [(C, D), (C, D)],
+            (0, 2): [(C, C), (C, D)],
+            (1, 1): [(D, D), (D, D)],
+            (1, 2): [(D, C), (D, D)],
+            (2, 2): [(C, C), (D, D)]
+        }
+
+        i.e. the key is a pair of player index numbers and the value, a list of
+        plays. The list contains one pair per turn in the round robin.
+        The dictionary contains one entry for each combination of players.
+
+    nplayers : integer
+        The number of players in the round robin
+
+    Returns
+    -------
+    list
+        The cooperation matrix (C) of the form:
+
+            [
+                [a, b, c],
+                [d, e, f],
+                [g, h, i],
+            ]
+
+        i.e. an n by n matrix where n is the number of players. Each row (i)
+        and column (j) represents an individual player and the the value Cij
+        is the number of times player i cooperated against opponent j.
     """
+    cooperation = [[0 for i in range(nplayers)] for j in range(nplayers)]
+    for players, actions in interactions.items():
+        p1_actions, p2_actions = zip(*actions)
+        p1_cooperation = p1_actions.count(Actions.C)
+        p2_cooperation = p2_actions.count(Actions.C)
```

Comment:
Ah yes! I added the variables in at the top of the file and then forgot to use them. Duh.